mne/viz/evoked.py

```diff
@@ -875,3 +880,142 @@ def plot_snr_estimate(evoked, inv, show=True):
     plt.draw()
     plt_show(show)
     return fig
+
+
+def _connection_line(x, fig, sourceax, targetax):
+    """Helper function to connect time series and topolots"""
+    from matplotlib.lines import Line2D
+    transFigure = fig.transFigure.inverted()
+    tf = fig.transFigure
+
+    (xt, yt) = transFigure.transform(targetax.transAxes.transform([.5, 0]))
+    (xs, _) = transFigure.transform(sourceax.transData.transform([x, 0]))
+    (_, ys) = transFigure.transform(sourceax.transAxes.transform([0, 1]))
+    return Line2D((xt, xs), (yt, ys), transform=tf, color='grey',
+                  linestyle='--', linewidth=1.5, alpha=.66, zorder=-1)
+
+
+def joint_plot(evoked, title='', picks=None, exclude=list(), show=True,
+               ts_args=dict(), topomap_args=dict()):
+    """Plot evoked data as butterfly plot and add topomaps for selected
+    time points.
+
+    Parameters
+    ----------
+    evoked : instance of Evoked
+        The evoked instance.
+    title : str
+        The title.
+    picks : array-like of int | None
+        The indices of channels to plot. If None show all.
+    exclude : list of str | 'bads'
+        Channels names to exclude from being shown. If 'bads', the
+        bad channels are excluded.
+    show : bool
+        Show figure if True.
+    ts_args : dict
+        A dict of `kwargs` that are forwarded to `evoked.plot` to
+        style the butterfly plot. `axes` and `show` are ignored.
+    topomap_args : dict
+        A dict of `kwargs` that are forwarded to `evoked.plot_topomap`
+        to style the topomaps. `axes` and `show` are ignored. If `times`
+        is not in this dict, automatic peak detection is used.
+
+    Returns
+    -------
+    fig : instance of matplotlib.figure.Figure | list
+        The figure object containing the plot. If `evoked` has multiple
+        channel types, a list of figures, one for each channel type, is
+        returned.
+
+    Notes
+    -----
+    .. versionadded:: 0.12.0
+    """
+    # channel selection
+    # simply create a new evoked object(s) with the desired channel selection
+    info = evoked.info
+    if exclude == "bads":
+        exclude = [ch for ch in info['bads'] if ch in info['ch_names']]
+    evoked = evoked.copy().drop_channels(exclude)
+    if picks is not None:
+        evoked = evoked.pick_channels(picks)
```

Comment:
I guess it only removes the intersection of the channels it's provided with, and the channels in the data set. Will fix.